content/04.workflows-and-software.md

```diff
@@ -80,36 +79,52 @@ Analysis workflows commonly rely on multiple software packages to generate final
+While these approaches do not integrate into reproducible workflows, they may be ideal for testing a tool to determine whether it is useful for your data before integration in your analysis.
+
+**Choosing a software management system**
+It is important to balance the time needed to learn to properly use a software management system with the needs of both the project and the researchers.
+Software management systems with large learning curves are less likely to be widely adopted among researchers with a mix of biological and computational backgrounds.
+In our experience, software management with conda nicely balances reproducibility with flexibility and ease of use.
+These trade-offs are best for research workflows under active development, where flexible software installation solutions that enable new analysis explorations or regular tool updates are critical.
+For production workflows that require maximal reproducibility, it is worth the larger investment required to use heavyweight systems.
+This is particularly true for advanced users who can more easily navigate the steps required for utilizing these tools.
+Container-based software installation via docker and singularity are common for production-level workflows, and Guix and Nix-based solutions are gaining traction.
+Importantly, the needs and constraints of a project can evolve over time, as may the system of choice.
+
 
 **Integrating software management within workflows**
-Workflow systems provide seamless integration with software management tools.
+Workflow systems provide seamless integration with a number of software management tools.
 Each workflow system requires different specification for initiation of software management, but typically requires about one additional line of code per step that requires the use of software.
 If the software management tool is installed locally, the workflow will automatically download and install the specified environment or container and use it for specified step.
 
-In our experience, the complete solution for using scientific software involves starting with a combination of interactive and exploratory analyses in IDEs and local conda installation to develop an analysis strategy and create an initial workflow.  This is then followed by workflow-integrated software management via conda, singularity, or docker for executing the resulting workflow on many samples.
+In our experience, the complete solution for using scientific software involves a combination of approaches. Interactive and exploratory analyses conducted in IDEs and jupyter notebooks (usually with local software installation with conda) are useful for developing an analysis strategy and creating an initial workflow. This is then followed by workflow-integrated software management via conda, singularity, or nixOS for executing the resulting workflow on many samples. This process is iterative as we build and extend our analyses.
```

Comment:
Re "This process is iterative as we build and extend our analyses" - grammatically questionable :). Maybe "We then iterate as we build and extend our analyses." Although come to think of it I'm not sure what we're iterating on. We then iteratively extend the workflow?